project/Build.scala

```diff
@@ -675,6 +675,73 @@ object Build {
     recur(lines)
   }
 
+  val shadedSourceGenerator = (Compile / sourceGenerators) += Def.task {
+    val s = streams.value
+    val cacheDir = s.cacheDirectory
+    val dest = (Compile / sourceManaged).value / "downloaded"
+    
+    val downloads = Seq(
+      "https://repo1.maven.org/maven2/com/lihaoyi/pprint_3/0.9.3/pprint_3-0.9.3-sources.jar",
+      "https://repo1.maven.org/maven2/com/lihaoyi/fansi_3/0.5.1/fansi_3-0.5.1-sources.jar",
+      "https://repo1.maven.org/maven2/com/lihaoyi/sourcecode_3/0.4.4/sourcecode_3-0.4.4-sources.jar",
```

Comment:
Using hard-coded URLs to `repo1.maven.org` is not friendly to environments that use a Maven proxy. Sometimes these environment outright prevent direct access to these URLs.

We should use `dependencyResolution.value.retrieve` with Maven coordinates instead, like for example here:
https://github.com/scala-js/scala-js/blob/8e82ab9c423c1cb9f70221cc16900848dec58a8f/sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPlugin.scala#L334-L339